src/lib/actions/BaseAction.ts

```diff
@@ -3,20 +3,53 @@ import ora, {Ora} from "ora";
 import chalk from "chalk";
 import inquirer from "inquirer";
 import { inspect } from "util";
-import {KeypairManager} from "../accounts/KeypairManager";
 import {createClient, createAccount} from "genlayer-js";
 import {localnet} from "genlayer-js/chains";
 import type {GenLayerClient, GenLayerChain} from "genlayer-js/types";
+import { ethers } from "ethers";
+import { writeFileSync, existsSync, readFileSync } from "fs";
+import { KeystoreData } from "../interfaces/KeystoreData";
 
 export class BaseAction extends ConfigFileManager {
-  protected keypairManager: KeypairManager;
   private spinner: Ora;
   private _genlayerClient: GenLayerClient<GenLayerChain> | null = null;
 
   constructor() {
     super();
     this.spinner = ora({text: "", spinner: "dots"});
-    this.keypairManager = new KeypairManager();
+  }
+
+  private async decryptKeystore(keystoreData: KeystoreData, attempt: number = 1): Promise<string> {
+    try {
+      const message = attempt === 1 
+        ? "Enter password to decrypt keystore:" 
+        : `Invalid password. Attempt ${attempt}/3 - Enter password to decrypt keystore:`;
+      const password = await this.promptPassword(message);
+      const wallet = await ethers.Wallet.fromEncryptedJson(keystoreData.encrypted, password);
+      return wallet.privateKey;
+    } catch (error) {
+      if (attempt >= 3) {
+        this.failSpinner("Maximum password attempts exceeded (3/3).");
```

Comment:
@epsjunior how is this preventing a bad actor to try again? Shouldn't the file be removed or something after x unsuccessful attempts?